MyQ Garage Opener

berg0319 opened the pull request.

brbeaird committed:
Merge pull request #2 from copy-ninja/master

Refresh fork

brbeaird committed:
Updated namespace to brbeaird
Removed light controller

brbeaird committed:
Renamed SmartApp to MyQLite

brbeaird committed:
MyQLite is born!

brbeaird committed:
Update README.md

brbeaird committed:
Fix issue with Alexa not being able to close the door (credit: gussery3)

brbeaird committed:
Fix handling of opening/closing status

brbeaird committed:
Fix naming for github integration

brbeaird committed:
Renaming for github integration

brbeaird committed:
Enable debugging for sensor changes

brbeaird committed:
Fix issue with Push toggle functionality

brbeaird committed:
Add more subscriptions to keep doors and sensors in sync
Add better logging for actions initiated from child door devices

brbeaird committed:
Add refresh to keep doors and sensors better in sync
Add better logging for events initiated by the door device

brbeaird committed:
Fix mode subscriptions to point to correct method.

brbeaird committed:
Update README.md

brbeaird committed:
Update README.md

brbeaird committed:
Correct last updated value

brbeaird committed:
Correct last updated value

brbeaird committed:
Add handling of null timestamps from sensors that haven't moved after 48(ish) hours.

brbeaird committed:
Add handling of null timestamps from stale sensors 
After 48(ish) hours, some sensors stop reporting data. This change prevents that empty data from overriding what we have stored.

brbeaird committed:
Extend sensor timestamp activity search back 7 days instead of 1

brbeaird committed:
Add tile to show the sensor associated with the door

brbeaird committed:
Add friendlier logging for door names
Add support for updating sensor tile in door devices

brbeaird committed:
Fix bug where doors 3 and 4 were always sync'ing to sensor #2.

brbeaird committed:
Update README.md

Fix readme typo

SANdood committed:
Added support for MultiSensor/Acceleration Sensor (#3)

* Updated for optional acceleration Sensor support

brbeaird committed:
Fixed bug where doors sometimes wouldn't update due to null sensor values

brbeaird committed:
Fixed issue where switch wasn't totally behaving as a switch (was specifically causing issues with SmartTiles).

brbeaird committed:
Ignore VGDO devices from MyQ

brbeaird committed:
Now supporting doors without needing sensors!
Added new non-sensor device type
Added optional PushButtons for both device types

brbeaird committed:
Update README.md

brbeaird committed:
Update README.md

brbeaird committed:
Update README.md

brbeaird committed:
Update README.md

brbeaird committed:
Renamed push-buttons to "Closer" and "Opener" for easier Alexa interaction

brbeaird committed:
Adding from ST integration

brbeaird committed:
Revert renaming no-sensor door device

brbeaird committed:
Adding Momentary Button Tile to repo

brbeaird committed:
Update README.md

brbeaird committed:
Added Alexa instructions to README

brbeaird committed:
Updated Readme links

brbeaird committed:
Update README.md

brbeaird committed:
Update README.md

brbeaird committed:
Simplified on/off behavior and added switch tile

brbeaird committed:
Updated date

brbeaird committed:
Updated switch indicator to use the normal switch icon

brbeaird committed:
Updated acceleration sensor default to closing

brbeaird committed:
Modified logging to be more clear when updating status

brbeaird committed:
Fix date typo

brbeaird committed:
Re-added light control
Fixed issue where MyQ sometimes returns duplicate doors
Fixed issue where acceleration sensors would sometimes cause false movement events

brbeaird committed:
Fixed dates
Fixed light copy/paste error

brbeaird committed:
Update README.md

brbeaird committed:
Added special thanks

brbeaird committed:
Merge branch 'master' of https://github.com/brbeaird/SmartThings_MyQ

brbeaird committed:
Updating icon to use the full mobile width and light bulb icon.

brbeaird committed:
Updated for MyQ API changes
Login now uses POST with additional headers
AppID's have changed

brbeaird committed:
Updated LastUpdated line

brbeaird committed:
Add icons

brbeaird committed:
Rename 3x

brbeaird committed:
Point to new icon location

brbeaird committed:
Update README.md

Fix backwards sensor/no sensor text in manual install section

brbeaird committed:
Update README.md

brbeaird committed:
Fixed issue where door icons would sometimes not update automatically based on status.

brbeaird committed:
Update README.md

ph4r committed:
Created new Device for IFTTT mail integration (#5)

brbeaird committed:
Added summary page
This page reports if device creation/removal was successful.

brbeaird committed:
Fix issue with subscriptions being cancelled when Update method is called without calling Initialize

brbeaird committed:
Fix virtual contact switch organization for github integration

brbeaird committed:
Added versioning header

brbeaird committed:
Added version warning message to setup summary page

brbeaird committed:
Fix heading bug introduced last commit

brbeaird committed:
Redesigned open/close button on no-sensor device

brbeaird committed:
Update README.md

brbeaird committed:
Fix buttons on sensor door device

brbeaird committed:
Update README.md

brbeaird committed:
Added debugging for blank desc

brbeaird committed:
Get version numbers via actual JSON endpoint instead of janky web scraping

brbeaird committed:
Add hub ID association when creating devices
This fixes bug where ST sometimes doesn't allow deletion of non-hub devices

brbeaird committed:
Cleaned up callout for version checking

brbeaird committed:
Clean up version check

brbeaird committed:
Added version cleanup to uninstall method

brbeaird committed:
Fix typo

brbeaird committed:
Updated device handlers to use new standard color schemes
Updated door opener to set status to "opening" before sending the command

brbeaird committed:
Added UK URL (experimental)

brbeaird committed:
Reverted last change

brbeaird committed:
Added SecurityToken to header

brbeaird committed:
Added token to header for PUT calls

brbeaird committed:
Updated door closed color to white

brbeaird committed:
Fixed bug where app install would fail if using a lamp module on its own and no doors

brbeaird committed:
Re-added LastActivity timestamp

brbeaird committed:
More bug fixing for lights-only setup
Updated versioning

brbeaird committed:
Fixed versioning and last-updated date

brbeaird committed:
Updated Readme with BTC link

brbeaird committed:
Update README.md

brbeaird committed:
Update README.md

brbeaird committed:
Update README.md

brbeaird committed:
Update README.md

brbeaird committed:
Added support for commercial door types.

brbeaird committed:
Added installerManifest.json

brbeaird committed:
Updated installerManifest.json
Updated icon color scheme

brbeaird committed:
Updated installerManifest.json

brbeaird committed:
Updated installerManifest.json

brbeaird committed:
Updated installerManifest.json

brbeaird committed:
Updated installerManifest.json

brbeaird committed:
Updated banner

brbeaird committed:
Update README.md

brbeaird committed:
Update README.md

brbeaird committed:
Added handling for null hub query

brbeaird committed:
Update README.md

brbeaird committed:
Delete virtual-contact-sensor-switch.groovy

brbeaird committed:
Update README.md

Add details on Classic app support only

brbeaird committed:
Changed door closed background color to standard blue

gregmora commented:
Hi brbeaird I would like to get in touch with you, can you please help me set up properly MyQ Lite app? is not working correctly and I ran a few times your instructions.

Thanks,

Jose

brbeaird commented:
I can't do much here. Feel free to create an issue under my fork https://github.com/brbeaird/SmartThings_MyQ/issues. Are you getting a particular error message?

joshualyon committed:
Added 'actuator' capability (#36)

Added the 'actuator' capability to the definition so the device handler can be used with SmartApps like the SharpTools.io dashboard which supports `Garage Door Control` and `Door Control` devices, but expects them to either be authorized along side the `contact` or `switch` capability... or will fall back to the `actuator` capability like this case.

brbeaird committed:
Update IDE URL

brbeaird committed:
Door limit and new app support
Increase max door limit to 8
Add partial support for new SmartThings app (requires reinstall)

brbeaird committed:
Fix door setup typos

brbeaird committed:
Re-added door control capability

brbeaird committed:
Update ping call
Health check is causing issues in Classic app - trying to fix that

brbeaird committed:
Remove health check capability
Causing way too many issues right now - will wait until more official documentation exists

brbeaird committed:
Fix settings refresh issue
When door list changes, ST does not always fully refresh the underlying setting. This fix explicitly validates the door setting list.

brbeaird committed:
Fix validated door bug for single door setups

brbeaird committed:
More single-door fixing (credit @Zhourj)
Issue #44

brbeaird committed:
And even more single-door fixing

brbeaird committed:
Bump version

brbeaird committed:
Band-aid fixes for door syncing issue #49
Add refresh button to door to force sync with contact sensor
Add logic to force sensor refresh at mode change and sunrise/sunset

brbeaird committed:
Add lock capability (issue #48 - credit @jrseliga)

brbeaird committed:
Fix errors for no-sensor door types (issue #51)

brbeaird committed:
Add lock door type

brbeaird committed:
Lock and refresh changes
Moved lock functionality into separate device handler
Removed redundant Door Control capability
Fixed issue in setup when light is selected with no doors
Added periodic refresh to keep door status in sync with sensors

brbeaird committed:
Fix hubId debugging

brbeaird committed:
Add uninstall method/fix event log spamming
Added automatic force-uninstall method. Honestly not sure why I didn't think of this years ago...
Fixed issue with DTH where refresh was spamming the event log even when the status did not change
Fixed SmartApp versioning typo

brbeaird committed:
Update README.md

Add tilt sensor warning
Point to ST Community Installer
Add lock door documentation

brbeaird committed:
Add code to actually force-delete child devices

brbeaird committed:
Merge branch 'master' of https://github.com/brbeaird/SmartThings_MyQ

brbeaird committed:
Push buttons now optional

brbeaird committed:
Add better valid hub detection (#50)

brbeaird committed:
Re-add Door Control (its absence was causing issues with older automations)

brbeaird committed:
Lock option UI and door action fixes (#54)
Move lock option to hidden advanced section
Fix issue where tapping door did not actually generate an action
Smooth out opening/closing transition states

brbeaird committed:
Update README.md

brbeaird committed:
Fix SmartApp version in installerManifest.json

brbeaird committed:
Update README.md

brbeaird committed:
Update README.md

brbeaird committed:
Update README.md

brbeaird committed:
Fix UserDeviceDetails casing #57

brbeaird committed:
Fix other getDevicesURL reference #57

brbeaird committed:
Update README.md

brbeaird committed:
Added more handling for null door description

rboy1 committed:
Correct put API URL

The API is case sensitive, correcting the device control (PUT) URL

rboy1 committed:
Improve authentication management (#59)

* Improve login management

Handle login expiration
Better authentication management
Reduce load on servers by logging only when necessary
Handle response codes from PUT and GET

* Indicate the code type

Helps while copy/pasting code into IDE

* Indicate the code type

* Indicate the code type

* Indicate the code type

brbeaird committed:
Bump versions and extend token life

Co-authored-by: Brian Beaird <brbeaird@gmail.com>
Co-authored-by: rboy1 <rboy1@users.noreply.github.com>

brbeaird committed:
Updated MyQAppId and headers

rboy1 committed:
Fix content type (#60)

* Fix content type

Dont' define content type, under normal operations it returns JSON but when there's an error (like invalid token or rejection) it returns a text/html which then causes an error with ST. This forces automatic detection and works best with ST

* Bump version

brbeaird committed:
Updating community installer version

brbeaird committed:
V3 - Major code refactoring and clean up (#69)

Removed acceleration functionality (not worth the added complexity and confused people)
Removed extra refreshes (hopefully these aren't actually needed anymore)
Improved SmartApp screens and setup flow
Optional push notifications when version updates are available
Optional push notifications when door open/close commands fail
Removes cap on number of doors. Behind the scenes, this also means sensor variables are fully
dynamic instead of hard-coded.

brbeaird committed:
Bump installerManifest.json versions for ST Community Installer

brbeaird committed:
Update README.md

brbeaird committed:
Fix issue with open push-button
Enable push notifications for door errors

brbeaird committed:
Fix version storage bug

brbeaird committed:
Update to support MyQ API V5 (#70)

brbeaird committed:
Update lock door

brbeaird committed:
Fix light functionality for V5
Clean up version checking logic

brbeaird committed:
Update README.md

brbeaird committed:
Update README.md

brbeaird committed:
Update README.md

brbeaird committed:
Update README.md

brbeaird committed:
Update README.md

brbeaird committed:
Remove brand from setup page

brbeaird committed:
Create FUNDING.yml

brbeaird committed:
Add battery capability and fix update messages
Add changelog

brbeaird committed:
Default sensor battery to 100

brbeaird committed:
Delete myq-lock-door.groovy

Remove duplicate file

brbeaird committed:
Update momentary-button-tile.groovy

brbeaird committed:
Add momentary button

brbeaird committed:
Move momentary button under same namespace
This helps the Community Installer

brbeaird committed:
Update refresh logic to use door status instead of contact

brbeaird committed:
Fix username/password casing

brbeaird committed:
Add virtual switch

brbeaird committed:
Convert momentary push-button to virtual switch

brbeaird committed:
Clarify switch text

brbeaird committed:
Update readme - remove momentary references

brbeaird committed:
Update readme

brbeaird committed:
Update door handler for new app

brbeaird committed:
Lower device URL casing

brbeaird committed:
Fix the sendCommand URL references

brbeaird committed:
Limit latest activity search to contact sensor events

brbeaird committed:
Update door icon in dashboard view

brbeaird committed:
Clean up door refresh logic

brbeaird committed:
Improve tracking of when sensors/buttons are used (fix #87)
More logging cleanup

brbeaird committed:
Fix latest activity bug

brbeaird committed:
Update CHANGELOG.md

brbeaird committed:
Add changelogs for DTH

brbeaird committed:
Clean up automation capabilities for sensor version
Limit to door control
Add custom DTH JSON presentation to source control

brbeaird committed:
Add no-sensor device handling for new app

brbeaird committed:
Fix dateModified typo

brbeaird committed:
Update README.md

brbeaird committed:
APIv6 (#129)

* Update auth and command flow for V6
* Add support for multiple accounts
* Handle bad 1% battery sensors
* Update README.md

brbeaird committed:
Update README.md

brbeaird committed:
Update README.md

brbeaird committed:
Fix no-sensor missing vid and add accountId fallback

brbeaird committed:
Token generator improvements
Automatically copy token to clipboard
Hold window open for errors

brbeaird committed:
Add AccountId fallback to light and sensor DTH

brbeaird committed:
Add handling for auto-copy token issues

brbeaird committed:
Oauth token generation (#146)

* Add methods to generate challenge and get token within SmartApp
* Modify UI flow for email/password auth
* Replace Virtual Switch with MyQ Action Type
* Update README.md

Co-authored-by: RBoy1 <3846367+rboy1@users.noreply.github.com>

brbeaird committed:
Update README.md

brbeaird committed:
Update README.md

brbeaird committed:
Fix close button typo

brbeaird committed:
Fix other close button syntax issue

rboy1 committed:
Version and suffix (#147)

Fixed version to sync change notes
v1 suffix not required

Co-authored-by: Brian Beaird <brbeaird@gmail.com>

brbeaird committed:
Update installerManifest.json for community installer

brbeaird committed:
Fix httpPost error

brbeaird committed:
Fixed issue where refresh was too frequent
Improved failed command error messages

brbeaird committed:
Move non-groovy files into misc folders
This should help avoid IDE errors

brbeaird committed:
Add redirect auth server URL option
Update default to https

brbeaird committed:
Add Readme deprecation warning

brbeaird committed:
Add link to new Edge driver

brbeaird committed:
Update README.md
